### DISCUSSION TITLE
Is there a way to manually clear or ignore Dynamic Link's getInitialLink after first use?

### DISCUSSION THREAD

**Discussion by kyle-ssg** [invertase/react-native-firebase#6603] [Q&A]
I think this is more of a dev experience issue than a bug, hence posting here to see if anyone has a way of working with it.

If I open a dynamic link from force closed app, ``getInitialLink`` will correctly return a link. However, every time I refresh the simulator (rr) ``getInitialLink`` will still resolve with the link from the first time.

My question is whether other people are experiencing this, do we just deal with it or do people have a way to clear/ignore already consumed initial links?

**Comment by mikehardy** [comment]
Hmm - I test links infrequently enough that I haven't bothered even handling the case in my work app personally.
If it is supposed to be consumed but is not, then the fix would be for the native code to store state on whether it was consumed or not, and only return it once, since refresh reboots the bundle but not the native code
A PR doing the same could work if you wanted to raise one
